src/types/object-validation-result.ts

```diff
@@ -0,0 +1,3 @@
+export type ObjectValidationResult<T> = {

```

Comment:
I know you didn't originally write it, but a JSDoc comment for what this type represents would probably go a long way (the syntax of the type is a bit funky). See the comment on the `ObjectValidatorInterface` for reference. Adding a short description and example of what it might look like would be awesome 👍